cobc/codegen.c

```diff
@@ -7098,7 +7098,9 @@ output_java_call (struct cb_call *p)
 	char *last_dot;
 	char *method_name;
 	const char *class_name;
+	char method_signature[256] = "(";
```

Comment:
For internal buffer sizes you have `COB_*_BUFF` definitions in `libcob/common.h` that you can use here as well. Typical signature length could exceed 256 without much effort I think (with long-enough Java package names). I'd go for `COB_NORMAL_BUFF` (2048) for now.